### PR TITLE
Apply styles to components. Close #17

### LINKED ISSUE #17: Passing in a component instead of a tag
```js
// styles.js

import { h } from 'hyperapp'
import style from 'picostyle'
const ps = style(h)

export const Nav = ps('nav')({
  position: 'fixed',
})

const Link = ({ url, text }) =>
  <a href={url}>{text}</a>

export const StyledLink = ps(Link)({
  background: 'red'
})
```
```js
// index.js

import { h, app } from 'hyperapp'
import { Nav, StyledLink } from './styles'

app({
  state: {},
  view: () => <Nav><StyledLink url='/' text='Home' /></Nav>
})
```

When I run the code above this is rendered:

```
<nav class='p0'>
  <a href='/'>Home</a>
</nav>
```

Elements created with a tag name as a string (like `ps('nav')`) get a style class added.. but any element created with tag as a component (like `ps(Link)`) does not get the class added to it.

Not sure if this is intentional or even possible to achieve.. but I know `h` accepts a string or a component as a the first argument.. and looking at the code [here](https://github.com/picostyle/picostyle/blob/master/src/index.js#L48) it looks as though that is all that we are doing with `tag`.

Any hints as to my wrong doing would be very much appreciated!

### PR DESCRIPTION
**Pull request by JorgeBucaran**
#17 
/cc @lukejacksonn

### CODE REVIEW

**Comment by lukejacksonn** [comment]
Wow.. that was fast 💯

File: src/index.js [hunk 1]
```diff
@@ -42,10 +42,15 @@ export default function (h) {
       var isDeclsFunction = typeof decls === "function"
       !isDeclsFunction && (parsed = parse(decls))
       return function (data, children) {
-        data = data || {}
-        isDeclsFunction && (parsed = parse(decls(data)))
-        data.class = ((data.class || '') + ' ' + parsed).trim()
-        return h(tag, data, children)
+        isDeclsFunction && (parsed = parse(decls(data = data || {})))
```
Comment (maxholman) [inline src/index.js]: `data` can still be undefined here if `isDeclsFunction` is falsy

Comment (JorgeBucaran) [reply src/index.js]: Whoops.

Comment (JorgeBucaran) [reply src/index.js]: Thanks @maxholman! Fixed 👍

**Comment by morishitter** [comment]
@JorgeBucaran Thanks for your quick response. Seems nice 👍

**Comment by JorgeBucaran** [comment]
When you get a chance merge & publish? 🏄👋😴

**Comment by morishitter** [comment]
I've just published v0.1.4. Thanks as always 😉

**Comment by lukejacksonn** [comment]
collab lvl.99999.. thank you all for being awesome 🤘🏼👨‍💻

**Pull request merged**